md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/AppDeploymentQueueAPI.java

```diff
@@ -46,9 +42,182 @@
+        AppDeploymentQueue returnedAppDeploymentQueue=new AppDeploymentQueue();
+        try{
+            String temp;
+            BufferedReader br = null;
+            String jsonfile="";
+            String homeDir = System.getProperty("user.home");
+            LOGGER.info("home directory" + homeDir);
+            br = new BufferedReader(new FileReader(homeDir+"/bdreappstore/store.json"));
+            while ((temp=br.readLine()) != null) {
+                jsonfile=jsonfile+temp;
+            }
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.configure(DeserializationConfig.Feature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            AppStore appStore = mapper.readValue(jsonfile, AppStore.class);
+            LOGGER.info("size of app catagory is "+appStore.getApplicationList().size());
+            com.wipro.ats.bdre.md.dao.jpa.AppDeploymentQueue appDeploymentQueue=appDeploymentQueueDAO.get(queueId);
+            LOGGER.info("AppDeploymentQueue  is "+appDeploymentQueue);
+            for (StoreJson storeJson:appStore.getApplicationList())
+            {
+                if (storeJson.getId().equals(appDeploymentQueue.getAppDomain()))
+                {
+                    LOGGER.info("no of apps in this catagory without updating is "+storeJson.getColumns().size());
+                    int alreadyExistedApp=0;
+                    AppValues addedApp=new AppValues();
+                    for (AppValues appValues:storeJson.getColumns())
+                    {
+                        if (appValues.getName().equals(appDeploymentQueue.getAppName()))
+                        {
+                            alreadyExistedApp=1;
+                        }
+                    }
+
+                    if (alreadyExistedApp==0)
+                    {
+                        UUID idOne = UUID.randomUUID();
+                        String temprory=storeJson.getId()+"/"+appDeploymentQueue.getAppName().toLowerCase().replace(" ","_");
+                        addedApp.setCategory("Category 1");
+                        addedApp.setDescription(appDeploymentQueue.getAppName());
+                        addedApp.setName(appDeploymentQueue.getAppName());
+                        addedApp.setIcon(temprory+"/analytic.png");
+                        addedApp.setLocation("bdreappstore-apps/"+temprory+".zip");
+                        addedApp.setId(idOne.toString());
+                    }
+                    storeJson.getColumns().add(addedApp);
+                    LOGGER.info("no of apps in this catagory after updating is "+storeJson.getColumns().size());
+
+                }
+            }
+            ObjectMapper mapper1 = new ObjectMapper();
+            try {
+               FileWriter fileOut = new FileWriter(homeDir+"/bdreappstore/store.json");
+                 mapper1.writeValue(new File(homeDir+"/bdreappstore/store.json"),appStore);
+            } catch (IOException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1166)